release.nix

```diff
@@ -58,30 +61,36 @@ let
       (builtins.attrValues ghcjs)
       (concatDepends ghc)
       (concatDepends ghcjs)
-      [command serverExeSkeleton iosSkeleton androidSkeleton]
+      (lib.optional reflex-platform.androidSupport androidSkeleton)
+      (lib.optional reflex-platform.iosSupport iosSkeleton)
+      [ command serverSkeletonExe serverSkeletonShell ]
     ];
-    command = local-self.command;
-    serverExeSkeleton = (import ./skeleton { obelisk = local-self; }).exe;
-    androidObelisk = import ./. (self-args // {
-      config.android_sdk.accept_license = true;
-    });
-    androidSkeleton = (import ./skeleton { obelisk = androidObelisk; }).android.frontend;
-    iosObelisk = import ./. (self-args // {
-      system = "x86_64-darwin";
-      iosSdkVersion = "10.2";
-    });
-    iosSkeleton = (import ./skeleton { obelisk = iosObelisk; }).ios.frontend;
+    command = obelisk.command;
+    skeleton = import ./skeleton { inherit obelisk; };
+    serverSkeletonExe = skeleton.exe;
+    # TODO fix nixpkgs so it doesn't try to run the result of haskell shells as setup hooks.
+    serverSkeletonShell = local-self.nixpkgs.runCommand "shell-safe-for-dep" {} ''
+      touch "$out"
+      echo "return" >> "$out"
+      cat "${skeleton.shells.ghc}" >> "$out"
+    '';
+    androidSkeleton = (import ./skeleton { inherit obelisk; }).android.frontend;
+    iosSkeleton = (import ./skeleton { inherit obelisk; }).ios.frontend;
```

Comment:
`system` needs to be overridden here

Reply:
It actually doesn't because we only include `iosSkeleton` for the platforms that cross compiler to iOS. `system` for us is the build platform, not host platform.

Reply:
Will this result in ios being built on a linux machine with a darwin remote builder?

Reply:
On Linux machine
```
nix-repl> x86_64-linux.profiled.iosSkeleton
error: attribute 'iosSkeleton' missing, at (string):1:1

nix-repl> x86_64-linux.profiled.iosSkeleton
error: attribute 'iosSkeleton' missing, at (string):1:1

nix-repl> x86_64-linux.unprofiled.iosSkeleton
error: attribute 'iosSkeleton' missing, at (string):1:1

nix-repl> x86_64-darwib.unprofiled.iosSkeleton
# starts getting a whole bunch of cabal2nix from darwin cache
```

So the key points are

1. `release.nix` doesn't meaningfully depend on `builtins.currentSystem`; the platforms are what they say
2. iOS builds are not done on Linux builders
3. Anyone can ask a Darwin remote builder to do an iOS build.